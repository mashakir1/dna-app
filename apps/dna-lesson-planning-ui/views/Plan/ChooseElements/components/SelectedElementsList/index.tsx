// Imports
import { Dispatch, SetStateAction } from "react";

// Components
import Dropbox from "../Dropbox";

// Types
import { Element } from "types";

// Styles
import styles from "./selectedElementsList.module.scss";

export interface SelectedElementsListProps {
  selectedIds: Array<string>;
  setSelectedIds: Dispatch<SetStateAction<Array<string>>>;
  selectedElements: Array<Element>;
  elementCrossOnClick: (id: string) => void;
}

const SelectedElementsList: (props: SelectedElementsListProps) => JSX.Element = ({
  selectedIds, setSelectedIds, selectedElements, elementCrossOnClick
}) => {
  return <div className={`${styles.selectedElementsList} flex gap-s`}>
    {selectedElements?.map((element, index) => <Dropbox
      key={element.id}
      id={index}
      selectedIds={selectedIds}
      setSelectedIds={setSelectedIds}
      element={element}
      elementCrossOnClick={elementCrossOnClick}
    />)}
  </div>;
};

export default SelectedElementsList;
