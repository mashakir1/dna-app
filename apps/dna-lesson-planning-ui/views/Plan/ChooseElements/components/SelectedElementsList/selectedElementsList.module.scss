.selectedElementsList {
  overflow-x: scroll;
  width: 100%;
}
