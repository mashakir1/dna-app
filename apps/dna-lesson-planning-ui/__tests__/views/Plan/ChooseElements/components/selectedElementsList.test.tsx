import { render, screen } from "@testing-library/react";
import SelectedElementsList from "views/Plan/ChooseElements/components/SelectedElementsList";
import { testElements } from "../mockData";

const dropboxMock = () => <div data-testid="Dropbox" />;
jest.mock(
  "views/Plan/ChooseElements/components/Dropbox",
  () => dropboxMock
);

describe("SelectedElementList", () => {
  it("renders an empty list", () => {
    const { container } = render(<SelectedElementsList
      selectedIds={[]}
      setSelectedIds={() => undefined}
      selectedElements={[]}
      elementCrossOnClick={() => undefined}
    />);

    expect(container).toMatchSnapshot();
  });

  it("renders a populated list", () => {
    const { container } = render(<SelectedElementsList
      selectedIds={[]}
      setSelectedIds={() => undefined}
      selectedElements={testElements}
      elementCrossOnClick={() => undefined}
    />);

    expect(container).toMatchSnapshot();
    expect(screen.queryAllByTestId("Dropbox")).toHaveLength(testElements.length);
  });
});
