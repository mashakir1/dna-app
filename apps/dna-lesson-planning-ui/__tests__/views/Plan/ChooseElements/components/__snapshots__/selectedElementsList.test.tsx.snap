// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedElementList renders a populated list 1`] = `
<div>
  <div
    class="selectedElementsList flex gap-s"
  >
    <div
      data-testid="Dropbox"
    />
    <div
      data-testid="Dropbox"
    />
    <div
      data-testid="Dropbox"
    />
  </div>
</div>
`;

exports[`SelectedElementList renders an empty list 1`] = `
<div>
  <div
    class="selectedElementsList flex gap-s"
  />
</div>
`;
